preinstall.js

```diff
@@ -59,13 +59,49 @@ switch (os.platform()) {
     break
 }
 
+function findMsBuild () {
+  var programmFiles
+
+  if (arch === 'x64') {
+    programmFiles = process.env['PROGRAMFILES(X86)']
+  } else {
+    programmFiles = process.env['PROGRAMFILES']
+  }
+
+  var possiblePaths = [
+    programmFiles + '/Microsoft Visual Studio/2017/BuildTools/MSBuild/15.0/Bin/msbuild.exe',
+    programmFiles + '/Microsoft Visual Studio/2017/Enterprise/MSBuild/15.0/Bin/msbuild.exe',
+    programmFiles + '/Microsoft Visual Studio/2017/Professional/MSBuild/15.0/Bin/msbuild.exe',
+    programmFiles + '/Microsoft Visual Studio/2017/Community/MSBuild/15.0/Bin/msbuild.exe',
+    programmFiles + '/MSBuild/14.0/Bin/MSBuild.exe',
+    programmFiles + '/MSBuild/12.0/Bin/MSBuild.exe',
+    process.env.WINDIR + '/Microsoft.NET/Framework/v4.0.30319/MSBuild.exe'
+  ]
+
+  for (var counter = 0; counter < possiblePaths.length; counter++) {
+    var possiblePath = possiblePaths[counter]
+
+    try {
+      fs.accessSync(path.resolve(possiblePath), fs.constants.R_OK | fs.constants.X_OK)
+      return possiblePath
+    } catch (error) {
+      // Binary not found checking next path
+    }
+  }
+
+  console.error('MSBuild not found')
+  console.info('You can run "npm install --global --production windows-build-tools" to fix this.')
```

Comment:
Should we make this `console.error` or `console.warn` to keep all output on stderr?

Reply:
My train of thought was, that this is more of a log level semantic, but reading the docs this does not seem to be the case. I'll change it to `console.error`.

Reply:
The console methods don't really carry any semantics in that regard (having only `stdout` and `stderr`) but are named as such for parity with the `console` in browsers :)